Sources/InternxtSwiftCore/Services/Network/NetworkFacade.swift

```diff
@@ -7,11 +7,45 @@
 
 import Foundation
 
+@available(macOS 10.15, *)
+
 public struct NetworkFacade {
-    private let apiUrl: String
-    private let bridgeUser: String
-    init(apiUrl: String , bridgeUser: String, userId: String) {
-        self.apiUrl = apiUrl
-        self.bridgeUser = bridgeUser
+    private let encrypt: Encrypt = Encrypt()
+    private let cryptoUtils: CryptoUtils = CryptoUtils()
+    private let mnemonic: String
+    private let upload: Upload
+    
+    public init(mnemonic: String, networkAPI: NetworkAPI, urlSession: URLSession? = nil){
+        self.mnemonic = mnemonic
+        self.upload = Upload(networkAPI: networkAPI, urlSession: urlSession)
+    }
+    
+    public func uploadFile(input: InputStream, encryptedOutput: URL, fileSize: Int, bucketId: String, progressHandler: @escaping ProgressHandler) async throws -> FinishUploadResponse {
+        // Generate random index, IV and fileKey
+        guard let index = cryptoUtils.getRandomBytes(32) else {
+            throw UploadError.InvalidIndex
+        }
+        
+        let fullHexString = cryptoUtils.bytesToHexString(index)
+        let hexIv = fullHexString.prefix(upTo: fullHexString.index(fullHexString.startIndex, offsetBy: 32))
```

Comment:
Is this reproducing correctly [this](https://github.com/internxt/sdk/blob/b0c7dec65b515b833f6bdba70de2170154a0decd/src/network/upload.ts#L33)?

If the offset is in bytes, and as each hex char is two bytes, you will be getting the IV from the index[0] to index[32] in bytes.

Reply:
this is correct, however i'm modifying it to slice the bytes array instead of the hex string, which is what we are actually doing in the sdk, that's why in the SDK we slice from 0 to 16, and here we slice from 0 to 32 (16 bytes)